actix-http-test/src/lib.rs

```diff
@@ -241,14 +241,27 @@ impl TestServer {
         response.body().limit(10_485_760).await
     }
 
+    /// Connect to WebSocket server at given path and configure WebsocketsRequest using provided function.
+    pub async fn ws_at_with_config<ConfFunc>(
+        &mut self,
+        path: &str,
+        config_func: ConfFunc,
+    ) -> Result<Framed<impl AsyncRead + AsyncWrite, ws::Codec>, awc::error::WsClientError>
+    where
+        ConfFunc: FnOnce(ws::WebsocketsRequest) -> ws::WebsocketsRequest,
```

Comment:
would prefer signature to receive `&mut WebsocketRequest` and return nothing

Reply:
Most methods on `WebsocketsRequest` like `.set_header` needs owned `self`.
Is it possible to perform such methods on `&mut` ?

Reply:
So they do, that's unfortunate. We'll leave this be for now.